pkgs/by-name/fo/foonathan-memory/package.nix

```
@@ -0,0 +1,62 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, fetchpatch
+, cmake
+, doctest
+}:
+
+let
+  version = "0.7-3";
+in
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "foonathan-memory";
+  version = builtins.replaceStrings [ "-" ] [ "." ] version;
+
+  src = fetchFromGitHub {
+    owner = "foonathan";
+    repo = "memory";
+    rev = "v${version}";
+    hash = "sha256-nLBnxPbPKiLCFF2TJgD/eJKJJfzktVBW3SRW2m3WK/s=";
+  };
```

In the spirit of `mkDerivation (finalAttrs { ... })`, this could be written something like:

```suggestion
stdenv.mkDerivation (finalAttrs: {
  pname = "foonathan-memory";
  version = builtins.replaceStrings [ "v" "-" ] [ "" "." ] finalAttrs.src.rev;

  src = fetchFromGitHub {
    owner = "foonathan";
    repo = "memory";
    rev = "v0.7-3";
    hash = "sha256-nLBnxPbPKiLCFF2TJgD/eJKJJfzktVBW3SRW2m3WK/s=";
  };
```

Or should `version` be written the same as upstream does?

```suggestion
stdenv.mkDerivation (finalAttrs: {
  pname = "foonathan-memory";
  version = "0.7-3";

  src = fetchFromGitHub {
    owner = "foonathan";
    repo = "memory";
    rev = "v${finalAttrs.version}";
    hash = "sha256-nLBnxPbPKiLCFF2TJgD/eJKJJfzktVBW3SRW2m3WK/s=";
  };
```

---

We want a change in the version to be reflected in the src, not the other way around. Also, the only relevant requirements here are that the version begins with a number and does not contain capital letters, as can be seen [here](https://github.com/NixOS/nixpkgs/blob/013f0a99e56eeb1cfb44764fc0d419306b84bbe4/pkgs/README.md#package-naming). I recommend just using `version = "0.7-3"`.